algorithms/unit_tests/TestStdAlgorithmsTeamAdjacentFind.cpp

```diff
@@ -0,0 +1,247 @@
+
+    switch (m_apiPick) {
+      case 0: {
+        auto it = KE::adjacent_find(member, KE::cbegin(myRowViewFrom),
+                                    KE::cend(myRowViewFrom));
+        Kokkos::single(Kokkos::PerTeam(member), [=]() {
+          if (it != KE::cend(myRowViewFrom)) {
+            m_destView(myRowIndex) = *it;
+          }
+
+          m_distancesView(myRowIndex) =
+              KE::distance(KE::cbegin(myRowViewFrom), it);
+        });
+        break;
+      }
+
+      case 1: {
+        auto it = KE::adjacent_find(member, myRowViewFrom);
+        Kokkos::single(Kokkos::PerTeam(member), [=]() {
+          if (it != KE::end(myRowViewFrom)) {
+            m_destView(myRowIndex) = *it;
+          }
+
+          m_distancesView(myRowIndex) =
+              KE::distance(KE::begin(myRowViewFrom), it);
+        });
+        break;
+      }
+
+      case 2: {
+        auto it = KE::adjacent_find(member, KE::cbegin(myRowViewFrom),
+                                    KE::cend(myRowViewFrom), m_binaryOp);
+        Kokkos::single(Kokkos::PerTeam(member), [=]() {
+          if (it != KE::cend(myRowViewFrom)) {
+            m_destView(myRowIndex) = *it;
+          }
+
+          m_distancesView(myRowIndex) =
+              KE::distance(KE::cbegin(myRowViewFrom), it);
+        });
+        break;
+      }
+
+      case 3: {
+        auto it = KE::adjacent_find(member, myRowViewFrom, m_binaryOp);
+        Kokkos::single(Kokkos::PerTeam(member), [=]() {
+          if (it != KE::end(myRowViewFrom)) {
+            m_destView(myRowIndex) = *it;
+          }
+
+          m_distancesView(myRowIndex) =
+              KE::distance(KE::begin(myRowViewFrom), it);
+        });
+        break;
+      }
+    }
+  }
+};
+
+template <class LayoutTag, class ValueType>
+void test_A(std::size_t numTeams, std::size_t numCols, int apiId) {
+  /* description:
+     use a rank-2 view randomly filled with values,
+     and run a team-level adjacent_find
+   */
+
+  // -----------------------------------------------
+  // prepare data
+  // -----------------------------------------------
+  // create a view in the memory space associated with default exespace
+  // with as many rows as the number of teams and fill it with random
+  // values from an arbitrary range.
+  auto [sourceView, sourceViewBeforeOp_h] = create_random_view_and_host_clone(
+      LayoutTag{}, numTeams, numCols,
+      Kokkos::pair{ValueType(5), ValueType(523)}, "sourceView");
+
+  // -----------------------------------------------
+  // launch kokkos kernel
+  // -----------------------------------------------
+  using space_t = Kokkos::DefaultExecutionSpace;
+  Kokkos::TeamPolicy<space_t> policy(numTeams, Kokkos::AUTO());
+
+  // to verify that things work, each team stores the result
+  // of its adjacent_find call, and then we check
+  // that these match what we expect
+  Kokkos::View<ValueType*> destView("destView", numTeams);
```

Comment:
you don't need this

Reply:
I use that to store actual values pointed by the iterator, and during comparison, I check if both Kokkos and std distances' are the same and if Kokkos and std values are the same. But sure - sourceView and sourceViewBeforeOp_h are the same, so that might be redundant, and only checking distances is sufficient.

Reply:
maybe i did not express myself clearly, but what i meant is that: 
- you allocate `destView` and pass it to the functor, but inside the functor is not used because adjacnet_find does not *write new values into another view*, it only reads values from view. 
- it is not even needed below when you check things 
So destView is not needed at all

Reply:
ah, if so, can you please change the name? in all the other tests destView refers to a destination view where we write things into

Reply:
Renamed to dataView and removed redundant comparison.